crypto/err/err.c

```diff
@@ -188,6 +188,8 @@ static int err_string_data_cmp(const ERR_STRING_DATA *a,
 static ERR_STRING_DATA *int_err_get_item(const ERR_STRING_DATA *d)
 {
     ERR_STRING_DATA *p = NULL;
+    if(err_string_lock == NULL)
```

Comment:
There's a couple of minor issues with this edit: lack of blank line between variable declaration and beginning of code (which was there before this edit) and missing space after `if` before `(`.

A smaller change could be to change the old line 192 from

```
   if (!CRYPTO_THREAD_read_lock(err_string_lock))
```

to

```
   if (err_string_lock == NULL || !CRYPTO_THREAD_read_lock(err_string_lock))
```

Reply:
The problem is that this also does not fix all potential issues with the teardown. I'd instead recommend building openssl with no-atexit configure option.